### PR TITLE
Fix GH-590: initial automated test

### PR DESCRIPTION
**Pull request by jwzimmer**
This is what I have for the initial automated test per https://github.com/LLK/scratch-www/issues/590#issuecomment-243812155. The initial test is to check that the URL in the Create link in the navbar is correct when the user is signed out (without navigating to the editor, and using selenium and tap, as discussed).
There are some questions I have about it:
- A better solution than brittle XPath (I can continue looking into this; first I wanted to get something working)
- I set it up with a driver for Chrome; should I enumerate through the other available drivers here? How does this work with saucelabs?
- I created an integration folder in the test folder for these kinds of automated tests, but I am open to changing the structure
- Integration with travis? (I think it should be simple?)
- I looked at your existing test files in scratch-www and tried to copy your conventions in terms of naming files, functions, etc., but obviously please let me know of any corrections
- Should I add selenium to [devDependencies](https://github.com/LLK/scratch-www/blob/develop/package.json)? I see that tap is in the list, but you guys use tap for unit tests as well, so I wanted to check what would be appropriate?

/cc @rschamp & @mewtaylor

### CODE REVIEW

**Comment by jwzimmer** [comment]
I will try to address the issues causing the Travis CI build to fail (I see the list in "Details).

**Comment by rschamp** [comment]
> Should I add selenium to devDependencies? I see that tap is in the list, but you guys use tap for unit tests as well, so I wanted to check what would be appropriate?

Yes, everything that's `require`d should be listed in `devDependencies`.  You can automatically add dev dependencies with `npm install <package name> --save-dev`

File: test/integration/check_navbar_links.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Checks that the links in the navbar on the homepage have the right URLs to redirect to
+ *
+ * Test cases: https://github.com/LLK/scratch-www/wiki/Most-Important-Workflows#Create_should_take_you_to_the_editor
+ */
+
+var tap=require('tap'),  
+    seleniumWebdriver = require('selenium-webdriver');
```
Comment (rschamp) [inline test/integration/check_navbar_links.js]: Style nit: each requirement should be on a new `var` declaration like
```javascript
var tap = require('tap');
var seleniumWebdriver = require('selenium-webdriver');
```

File: test/integration/check_navbar_links.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/*
+ * Remove question comments after resolving them
+ */
+//how to generalize for other browsers... how will this work in saucelabs? Do I need to iterate through the drivers for each browser here?
+//chrome driver
```
Comment (rschamp) [inline test/integration/check_navbar_links.js]: Using Sauce Connect with webdriver: https://wiki.saucelabs.com/display/DOCS/Updating+Test+Scripts+to+Use+Sauce+Connect
Here is how Travis gets set up with SauceLabs: https://docs.travis-ci.com/user/sauce-connect/

To simplify running these tests locally as well as on Travis, maybe we should just always run them in Sauce?

File: test/integration/check_navbar_links.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+//var createLinkSignedOut = navbarElement.findElement(seleniumWebdriver.By.xpath('//li[@class="link create"]/a'))
+//^^this doesn't work to force findElement to look in the scope of navbarElement
+
+//find the create link within the navbar
+//the create link depends on whether the user is signed in or not (tips window opens)
+
+/*
+ * Remove question comments after resolving them
+ */
+//this xpath is fragile, can i look up by successive attributes instead?
```
Comment (rschamp) [inline test/integration/check_navbar_links.js]: I think using the fragile XPath is a necessary evil of using Selenium this way. When we change the DOM structure, we will have to update the test, but this is frequently the case with refactors even when not using Selenium.

File: test/integration/check_navbar_links.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+tap.test('checkCreateLinkWhenSignedOut', function (t) {
+    var createLinkSignedOut = driver.findElement(seleniumWebdriver.By.xpath('//div[@id="navigation"]/div[@class="inner"]/ul/li[@class="link create"]/a'));
+    createLinkSignedOut.getAttribute('href').then( function (href) {
+        t.equal('https://scratch.ly/projects/editor/?tip_bar=home', href);
```
Comment (rschamp) [inline test/integration/check_navbar_links.js]: Is this really the href? We shouldn't be specifying the domain name, it should be something like `/projects/editor/?tip_bar=home`.

File: test/integration/check_navbar_links.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+//open scratch.ly in a new instance of the browser
+driver.get('https://scratch.ly');
+
+/*
+ * Remove question comments after resolving them
+ */
+//find the navbar
+//var navbarElement = driver.findElement(seleniumWebdriver.By.id("navigation"))
+//var createLinkSignedOut = navbarElement.findElement(seleniumWebdriver.By.xpath('//li[@class="link create"]/a'))
+//^^this doesn't work to force findElement to look in the scope of navbarElement
```
Comment (rschamp) [inline test/integration/check_navbar_links.js]: Does `contains` work? `//li[contains(@class, 'link') and contains(@class, 'create')]/a`?

File: test/integration/check_navbar_links.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+var tap=require('tap'),
+    seleniumWebdriver = require('selenium-webdriver');
+
+/*
+ * Remove question comments after resolving them
+ */
+//how to generalize for other browsers... how will this work in saucelabs? Do I need to iterate 
+//through the drivers for each browser here?
+//chrome driver
```
Comment (rschamp) [inline test/integration/check_navbar_links.js]: Eventually maybe we want to set the browser through an environmental variable and then iterate through running all the tests by switching the variable?  For now I think just one browser is ok.

File: package.json [hunk 7]
```diff
@@ -82,6 +82,7 @@
     "sass-lint": "1.5.1",
     "sass-loader": "2.0.1",
     "scratchr2_translations": "git://github.com/LLK/scratchr2_translations.git#master",
+    "selenium-webdriver": "^2.44.0",
```
Comment (rschamp) [inline package.json]: Please remove the `^` to match the other dependencies. Info on what it means: https://docs.npmjs.com/getting-started/semantic-versioning. And you can set up your npm to never include it with `npm config set save-prefix=''`: https://docs.npmjs.com/misc/config#save-prefix

File: test/integration/check_navbar_links.js (hunk 4 above)
Comment (mewtaylor) [inline test/integration/check_navbar_links.js]: In the code, it's set to the relative path, it appears

Comment (rschamp) [inline test/integration/check_navbar_links.js]: The browser probably supplies the full URL when Selenium looks it up, so the test should check if it ends with the path.

File: test/integration/check_navbar_links.js (hunk 5 above)
Comment (jwzimmer) [inline test/integration/check_navbar_links.js]: `contains` works, now using `contains`, thanks.

**Comment by jwzimmer** [comment]
...I think the way I implemented looking only at the href rather than all the URL is bad (too convoluted); I will change it. : )

**Comment by jwzimmer** [comment]
(We will address https://github.com/LLK/scratch-www/pull/920#discussion_r78991538 later (setting up running the tests with travis/ sauce).)

**Pull request merged**